modules/fedimint-ln-server/src/lib.rs

```diff
@@ -1036,9 +1035,31 @@ impl Lightning {
         dbtx: &mut ModuleDatabaseTransaction<'_>,
         gateway: LightningGateway,
     ) {
+        // Garbage collect expired gateways (since we're already writing to the DB)
+        self.delete_expired_gateways(dbtx).await;
+
         dbtx.insert_entry(&LightningGatewayKey(gateway.node_pub_key), &gateway)
```

Comment:
Let's make a issue for this. We shouldn't be storing gateways by their `node_pub_key`. It is possible that two gateways use the same lightning node.